appinventor/buildserver/src/com/google/appinventor/buildserver/resources/runtime.scm

```diff
@@ -434,12 +443,37 @@
                          registeredComponentName eventName)
                        #f))))
 
+       ;; For the HandlesEventDispatching interface
+       (define (dispatchMultiEvent eventName :: java.lang.String
```

Comment:
Just wanted you to know that I haven't forgotten about this.  I just haven't gotten much time to take a close look.  I have been thinking about approaches that might simplify things for you but I need to think it through.  I do have a clarifying question, though.  When you refer to multi-events here do you mean the same as what you've called multi-generic in other places?

I'll point out that you say that the "multi-event will also be dispatched ...".  I hope that, as discussed, you plan to only allow for a single handler to be dispatched for any given event.